arch/arm/boot/dts/overlays/hifiberry-dac8x-overlay.dts

```diff
@@ -43,6 +48,7 @@
 		__overlay__ {
 			compatible = "hifiberry,hifiberry-dac8x";
 			i2s-controller = <&i2s_clk_producer>;
+			hasadc-gpio = <&gpio 5 GPIO_ACTIVE_HIGH>;
```

Comment:
I don't know the details of the hardware, but it looks strange to have a detection mechanism where the "detected" value is the same as the default with nothing connected - GPIO 5 pulls high by default, and this overlay doesn't change that.